src/main/webapp/app/overview/exercise-details/course-exercise-details.component.ts

```diff
@@ -118,9 +124,53 @@ export class CourseExerciseDetailsComponent implements OnInit, OnDestroy {
         this.exercise.participationStatus = participationStatus(this.exercise);
         this.isAfterAssessmentDueDate = !this.exercise.assessmentDueDate || moment().isAfter(this.exercise.assessmentDueDate);
         this.exerciseCategories = this.exerciseService.convertExerciseCategoriesFromServer(this.exercise);
+        if (this.exercise.type.toString() === 'programming') {
```

Comment:
No need for string comparison. Better do: `this.exercise.type === ExerciseType.PROGRAMMING`